### PR TITLE
GRID-364: Treeview: Do not filter parent rows

### PR DESCRIPTION
**Pull request by joneit**
Paired PR with https://github.com/openfin/hyper-analytics/pull/30

To run this you would have to:
* use the hyper-analytics in https://github.com/openfin/hyper-analytics/pull/30
* uncomment the line you commented-out in behaviors/JSON.js:
```javascript
this.setGlobalFilter(this.getNewFilter());
```

### CODE REVIEW

File: demo/js/demo.js [hunk 1]
```diff
@@ -1254,7 +1254,7 @@ window.onload = function() {
     }
 
     function resetFilter() {
-        grid.setGlobalFilter(Hypergrid.behaviors.Behavior.prototype.getNewFilter.call(grid.behavior));
+        grid.setGlobalFilter(grid.behavior.getNewFilter());
```
Comment (joneit) [inline demo/js/demo.js]: semantically neutral simplification

File: src/dataModels/JSON.js [hunk 2]
```diff
@@ -832,7 +831,12 @@ function reselectGridRowsBackedBySelectedDataRows() {
 
 function getDataSourceName(name) {
     name = analytics[name].prototype.$$CLASS_NAME || name;
-    return name.replace(/^Data(Source|Node)/, '').toLowerCase() || 'source';
+    if (/^DataSource\w+Filter$/.test(name)) {
+        name = 'filter';
```
Comment (joneit) [inline src/dataModels/JSON.js]: _Explanation:_ Any data source with the word "Filter" in it becomes `sources.filter`

File: src/dataSources/DataSourceOrigin.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 var DataSourceBase = require('./DataSourceBase');
-var headerify = require('hyper-analytics').util.headerify;
+var headerify = require('../Shared.js').analytics.util.headerify;
```
Comment (joneit) [inline src/dataSources/DataSourceOrigin.js]: @Dwaynekj This was the cause of your hyper-analytics reload issue.

File: add-ons/tree-view.js [hunk 4]
```diff
@@ -115,10 +115,20 @@ TreeView.prototype = {
...
             dataModel = behavior.dataModel,
             dataSource = dataModel.sources.treeview,
-            joined = dataSource.setRelation(options),
             state = behavior.getPrivateState(),
+            filterDataSource = this.grid.behavior.dataModel.getGlobalFilterDataSource(),
+            filter = filterDataSource.get();
+
+        // clear filter before setRelation call
+        filterDataSource.set(undefined);
+        dataModel.applyAnalytics();
```
Comment (Dwaynekj) [inline add-ons/tree-view.js]: should be behavior.applyAnalytics but I will fix when I do the big merge

**Pull request merged**